src/sympc/protocol/aby3/aby3.py

```diff
@@ -277,22 +294,133 @@ def bit_injection(x: MPCTensor, session: Session, ring_size: int) -> MPCTensor:
 
         args = [[share, str(ring_size)] for share in x.share_ptrs]
 
-        decompose = parallel_execution(ABY3.local_decomposition, session.parties)(args)
+        decomposed_shares = parallel_execution(
+            ABY3.local_decomposition, session.parties
+        )(args)
 
         # Using zip for grouping on pointers is compute intensive.
-        x1_sh = []
-        x2_sh = []
-        x3_sh = []
+        x1_share = []
+        x2_share = []
+        x3_share = []
 
-        for sh in decompose:
-            x1_sh.append(sh[0].resolve_pointer_type())
-            x2_sh.append(sh[1].resolve_pointer_type())
-            x3_sh.append(sh[2].resolve_pointer_type())
+        for share in list(
+            map(lambda x: x[0].resolve_pointer_type(), decomposed_shares)
```

Comment:
Q: If you drop the `resolve_pointer_type()` from here it does not work? (I think it should work since we know with what type we are deadling -- a list of RST Shares)

Reply:
I think it got a error while development at initial stages , I will try it out now.

Reply:
I remember now , when we subscript a pointer, it returns a any pointer, which we again subscript at next step, so I did resolve pointer type george